src/Symfony/Component/HttpKernel/Tests/HttpKernelTest.php

```diff
@@ -136,6 +138,32 @@ public function getStatusCodes()
         );
     }
 
+    /**
+     * @dataProvider getSuccessfulStatusCodes
+     */
+    public function testHandleWhenAnExceptionIsHandledWithASpecificStatusCode($expectedStatusCode)
+    {
+        $dispatcher = new EventDispatcher();
+        $dispatcher->addListener(KernelEvents::EXCEPTION, function (GetResponseForExceptionEvent $event) use ($expectedStatusCode) {
+            $event->setAllowCustomResponseCode(true);
+            $event->setResponse(new Response('', $expectedStatusCode));
+        });
+
+        $kernel = $this->getHttpKernel($dispatcher, function () { throw new \RuntimeException(); });
+        $response = $kernel->handle(new Request());
+
+        $this->assertEquals($expectedStatusCode, $response->getStatusCode());
+    }
+
+    public function getSuccessfulStatusCodes()
```

Comment:
Not just successful ones, just one that is different from the one Symfony would set by default.